graalpy/graalpy-scripts-debug-guide/README.md

```diff
@@ -0,0 +1,244 @@
+
+❹ The `dap` dependency provides a remote debugger for GraalPy that we can use when Python code is embedded in a Java application.
+
+## 4.2 Adding a Python script
+
+We can just include simple Python scripts in our resources source folder.
+In this example, the script contains a function that uses the Python standard library to compute the similarity between two files.
+
+`src/main/resources/compare_files.py`
+```python
+import polyglot  # pyright: ignore
+
+from difflib import SequenceMatcher
+from os import PathLike
+
+
+@polyglot.export_value # ①
+def compare_files(a: PathLike, b: PathLike) -> float:
+    with open(a) as file_1, open(b) as file_2: 
+        file1_data = file_1.read() 
+        file2_data = file_2.read() 
+        similarity_ratio = SequenceMatcher(None, file1_data, file2_data).ratio()
+        return similarity_ratio
+```
+
+❶ The only GraalPy-specific code here is this `polyglot.export_value` annotation, which makes the function accessible by name to the Java world.
+
+## 4.2.1 Working with GraalPy in VSCode
+
+You can use [pyenv](https://github.com/pyenv/pyenv) or [pyenv-win](https://github.com/pyenv-win/pyenv-win) with the [Python extensions](https://marketplace.visualstudio.com/items?itemName=ms-python.python) in VSCode to setup and use GraalPy during development.
+You can than edit and debug your Python files using the standard Python tooling.
+
+![Gif animation of installing GraalPy with pyenv](./graalpy-vscode-pyenv.gif)
+![Gif animation of using GraalPy in VSCode](./graalpy-vscode-select.gif)
+![Gif animation of debugging with GraalPy in VSCode](./graalpy-vscode-debug.gif)
+
+## 4.3 Creating a Python context
+
+GraalVM provides Polyglot APIs to make starting a Python context easy.
+We create the Python context in the JavaFX `start` method.
+We also override the `stop` method to close the context and free any associated resources.
+
+`App.java`
+```java
+public class App extends Application {
+    private Context context;
+
+    @Override
+    public void stop() throws Exception {
+        context.close();
+        super.stop();
+    }
+
+    @Override
+    public void start(Stage stage) {
+        context = Context.newBuilder("python")
+            .allowIO(IOAccess.newBuilder() // ①
+                            .fileSystem(FileSystem.newReadOnlyFileSystem(FileSystem.newDefaultFileSystem()))
+                            .build())
+            .allowPolyglotAccess(PolyglotAccess.newBuilder() // ②
+                            .allowBindingsAccess("python")
+                            .build())
+            // These are all the options we need to run the app
+```
+
+❶ By default, GraalPy will be sandboxed completely, but our script wants to access files.
+Read-only access is enough for this case, so we grant no more.
+
+❷ Our script exposes the `compare_files` function by name to the Java world.
+We explicitly allow this as well.
+
+## 4.3 Calling the Python script from Java
+
+`App.java`
+```java
```

Comment:
If I want to follow just the guide without looking at the reference solution, then this snippet is not enough. In other guides like the Micronaut/Spring guide we included even the large html listing just so that the guide itself is self-contained. We should probably be consistent in this?

Reply:
Good point, I'll expand the readme